test/schemes/fluid/rhs.jl

```diff
@@ -132,21 +139,34 @@
                                                           pressure_acceleration=nothing,
                                                           density_calculator=density_calculator,
                                                           smoothing_length, 0.0)
+
+                system_iisph = ImplicitIncompressibleSPHSystem(fluid, smoothing_kernel,
+                                                               smoothing_length, 1000.0,
+                                                               time_step=0.001)
+
                 n_particles = TrixiParticles.nparticles(system_edac)
 
                 # Overwrite `system.pressure` because we skip the update step
                 system_wcsph.pressure .= fluid.pressure
-                @testset "`$(nameof(typeof(system)))`" for system in (system_wcsph,
-                                                            system_edac)
+                system_iisph.pressure .= fluid.pressure
+                if density_calculator isa TrixiParticles.SummationDensity
+                    systems = (system_wcsph, system_edac, system_iisph)
+                else
+                    systems = (system_wcsph, system_edac)
+                end
+                @testset "`$(nameof(typeof(system)))`" for system in systems
                     u = fluid.coordinates
                     if density_calculator isa SummationDensity
                         # Density is stored in the cache
                         v = fluid.velocity
```

Comment:
I think the comment `# Density is stored in the cache` is now very confusing. How about changing the order?
```julia
                        # Density is stored in the cache
                        v = fluid.velocity
                        if system isa WeaklyCompressibleSPHSystem
                            system.cache.density .= fluid.density
                        elseif system isa EntropicallyDampedSPHSystem
                            # Pressure is integrated
                            system.cache.density .= fluid.density
                            v = vcat(fluid.velocity, fluid.pressure')
                        else
                            system.density .= fluid.density
                        end